### PR TITLE
More readable Contribution guide

### PR DESCRIPTION
**Pull request by CommanderStorm**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

This PR addresses a few issues I had when reading the contribution guide.
Among them:
- made sure that all technical terms have he proper code-block
- refactoring the maintainer-only procedures to be more compact by including `<details>` blocks
- added a rationale why a discussion is prefered
- added a section on `I'd like to work on an issue. How do I do that?`

Refactoring `Can I create a pull request for Uptime Kuma?` to give more actionable advice on how to create new monitoring/notification types was pulled into another PR for better reviewability: https://github.com/louislam/uptime-kuma/pull/4545

## Type of change

Please delete any options that are not relevant.

- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -154,25 +166,23 @@ npm run start-server-dev
 
 It binds to `0.0.0.0:3001` by default.
 
-It is mainly a socket.io app + express.js.
-
-express.js is used for:
+It is mainly a `socket.io`-app, but includes `express.js` to serve:
```
Comment (chakflying) [inline CONTRIBUTING.md:169]: I think a more accurate description would be

The backend is an `express.js` server with `socket.io` integrated. It uses `socket.io` to communicate with clients, and most server logic is encapsulated in the `socket.io` handlers. `express.js` is also used to serve:

**Review by chakflying — COMMENTED** [review]
I guess I don't consider myself to be a maintainer since I don't merge any code. But I'm fine with any title I guess.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -66,11 +63,20 @@ I ([@louislam](https://github.com/louislam)) have the final say. If your pull re
...
+
+We have found that assigning people to issues is management-overhead that we don't need.
+A short comment that you want to try your hand at this issue is appreciated to save other devs time.
+If you come across any issue, feel free to leave a comment with what you are stuck with.
 
 ### Recommended Pull Request Guideline
 
-Before deep into coding, discussion first is preferred. Creating an empty pull request for discussion would be recommended.
+Before diving deep into coding, having a discussion creating an empty pull request for discussion is prefered.
+The rationale behind this is that we might know which PRs do some work into the same direction and can point out pitfalls / conflicts.
```
[the remark this answers is not in the document]
Comment (CommanderStorm) [reply CONTRIBUTING.md:79]: Think this needs a "by" after "can help"

```suggestion
The rationale behind this is that we can align the direction and scope of the feature to eliminate any conflicts with existing and planned work, and can help by pointing out any potential pitfalls.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (CommanderStorm) [reply CONTRIBUTING.md:169]:
```suggestion
The backend is an `express.js` server with `socket.io` integrated.
It uses `socket.io` to communicate with clients, and most server logic is encapsulated in the `socket.io` handlers.
`express.js` is also used to serve:
```

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chakflying) [reply CONTRIBUTING.md]: Sounds good to me.

**Pull request merged**